### ISSUE TITLE
Crash reading >= 3D array in TTree via MakeClass in Windows ROOT6 compilation

### ISSUE THREAD

**Issue by pav-rez** [root-project/root#8833]
Dear ROOT developers,

we have met with a problem reading more than 2-dimensional arrays in TTrees using MakeClass. The problem appears only on the **Windows ROOT 6 version** (same code works on ROOT 6 on Linux, as well as on ROOT 5 in Windows). The code to reproduce is simple:

A tree is created by:
```
TFile f("newtree.root", "NEW");
TTree t("MyTree", "Test tree");
double x[5][5][5];
t.Branch("x", x, "x[5][5][5]");
t.Fill();
t.Write();
f.Close();
```
Then produce MakeClass from the tree: ```MyTree->MakeClass()```.

Finally, try to read that tree via the class MyTree:
```
.L MyTree.C
MyTree t;
t.Loop(); // crashes with bad-memory-allocation message (or silently)
t.b_x->GetAddress(); // also crashes
```

No problem is seen when using 2D or 1D arrays, nor simple numbers.

Thanks, Pavel

**Comment by bellenot** [comment]
I can reproduce the issue, and also with 2D array. The problem comes from Cling. We will investigate

**Comment by bellenot** [comment]
Closing this issue. This is fixed in 6.32 and master
